Frameworks/Foundation/NSURLResponse.mm

```diff
@@ -89,8 +94,22 @@ - (id)_initWithURL:(NSURL*)url
         _url.attach([url copy]);
         _mimeType.attach([mimeType copy]);
         _textEncodingName.attach([textEncodingName copy]);
-        if (nil != fileName) {
-            _suggestedFilename.attach([fileName copy]);
+
+        if (fileName && ![fileName isEqualToString:s_unknownFileName]) {
+            woc::unique_cf<CFStringRef> identifier {
+                _UTTypeCreatePreferredIdentifierForTag(static_cast<CFStringRef>(@"public.mime-type"),
+                                                       static_cast<CFStringRef>(mimeType),
+                                                       nullptr)
+            };
+            StrongId<NSString> extension {
+                static_cast<NSString*>(
+                    _UTTypeCopyPreferredTagWithClass(identifier.get(), static_cast<CFStringRef>(@"public.filename-extension")))
+            };
+
+            // If the filename already has the extension do not append
```

Comment:
>// If the filename already has the extension do not append [](start = 12, length = 58)

more like: // If the filename already has the expected extension (based on MIME type_ do not append